### DISCUSSION TITLE
Query parameter list example in docs throws AssertionError: `Query` default value cannot be set in `Annotated` for 'q'. Set the default value with `=` instead.

### DISCUSSION THREAD

**Discussion by epicwhale** [fastapi/fastapi#9283] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import Annotated

@router.get("/test-list")
async def read_items(q: Annotated[list[str] | None, Query()] = None):
    query_items = {"q": q}
    return query_items
```

### Description

When I try to create a query param that is a list[] but also optional and defaults to a None, I get a "AssertionError: `Query` default value cannot be set in `Annotated` for 'q'. Set the default value with `=` instead" - but as you can see above, the default value is already set using the = syntax.

I thought this was a problem with my code, but I tried running the example from the fastapi docs here - https://fastapi.tiangolo.com/tutorial/query-params-str-validations/#query-parameter-list-multiple-values and it throws the same error.

```
 |     app.include_router(courts_table_router, prefix="/table", tags=["table"])
 |   File "/usr/local/lib/python3.11/site-packages/fastapi/applications.py", line 427, in include_router
 |     self.router.include_router(
 |   File "/usr/local/lib/python3.11/site-packages/fastapi/routing.py", line 782, in include_router
 |     self.add_api_route(
 |   File "/usr/local/lib/python3.11/site-packages/fastapi/routing.py", line 600, in add_api_route
 |     route = route_class(
 |             ^^^^^^^^^^^^
 |   File "/usr/local/lib/python3.11/site-packages/fastapi/routing.py", line 446, in __init__
 |     self.dependant = get_dependant(path=self.path_format, call=self.endpoint)
 |                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 |   File "/usr/local/lib/python3.11/site-packages/fastapi/dependencies/utils.py", line 303, in get_dependant
 |     type_annotation, depends, param_field = analyze_param(
 |                                             ^^^^^^^^^^^^^^
 |   File "/usr/local/lib/python3.11/site-packages/fastapi/dependencies/utils.py", line 387, in analyze_param
 |     assert field_info.default is Undefined or field_info.default is Required, (
 |                                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 | AssertionError: `Query` default value cannot be set in `Annotated` for 'q'. Set the default value with `=` instead.
```

### Operating System

macOS

### FastAPI Version

0.95.0

### Python Version

3.11.2

**Comment by tlinhart** [comment]
Reported earlier [here](https://github.com/tiangolo/fastapi/discussions/9279).
